Replacing ExampleScript with SimplePlaythrough

RaduDumitriu-Dev opened the pull request:
Implemented SimplePlaythrough MonoBehaviour, acting as a simple controller of a Playthrough class instance and featuring standalone functionality.

Users can opt to use the Standalone features, which hook into simple Player and Actor MonoBehaviours, which will be updated moving forward - or opt out and manually use the exposed public functions the SimplePlaythrough monobehaviour has.

With this change comes included two new scenes featuring identical functionality, acting as example of both Standalone and External implementations.

bensalilijames reviewed (changes requested):
- Remove `_standalone`, `_player`, `_actor` from `SimplePlaythrough`
- Instead have `SimplePlaythrough` as a singleton, created on `Awake()`, so other components can access it with `SimplePlaythrough.Instance`
- Then `SimpleCharismaActor` and `SimpleCharismaPlayer` can bind directly to `SimplePlaythrough.Instance` in their respective `Awake()`s without the user having to set up any linkage between the two
- It's super extendable, e.g. to write another kind of component like something to handle metadata, someone would only need to bind to `SimplePlaythrough.Instance` and they're done
- Downside is you can only have one playthrough active at a time (singleton). But in reality there's not yet been a situation where we've had to use two or more and would be surprised if so! If that's really an issue for someone they can always drop-down to the lower level `Playthrough` class anyway

RaduDumitriu-Dev commented:
Hey Ben - I addressed the feedback of converting the SimplePlaythrough to a singleton. I do think there are some downsides to this approach that may be worth considering when looking at more complex systems like the plug and play:

- singleton can be requested inside of Playthrough entities, which means an object in service to the playthrough can start commanding it, which nullifies that hierarchy. (Players do have some ability to command the SimplePlaythrough ofc, but i dont think Actors should). Any exposed functionality for other systems would carry the risk of playthrough entitties calling them as well.
- pinging for the singleton on Awake is alright, but I think users would want that specific control of Registering/Unregistering be it at edittime or runtime as pinging every x seconds is not ideal (note: `Awake `functions trigger in any order, so the playthrough can have its `awake `called after the `_actor `and `_player`, works better in unreal as I think the game mode script takes priority over any world entities). 
- i think message handling would be a bit awkward as well, registering to it directly means duplicating code in the simpleplaythrough, and exposing the playthrough event is out of the question imo (don't want users doing SimplePlaythrough.Playthrough.Disconnect() for example, without the Simple knowing)

RaduDumitriu-Dev commented:
Feedback from previous review should be addressed. I've also updated to accomodate for the latest SDK updates. Bumping this for review
